### PR TITLE
allocate memory in segment.c

### PR DESCRIPTION
**Pull request by akshayben**
Allocation of Memory Segment
- allocate and free memory created

Signed-off-by: akshayben <akshayben@ibm.com>

### CODE REVIEW

**Comment by akshayben** [comment]
The PR will only go in after https://github.com/TobiAjila/openj9/pull/17
The commit is tested on top of restructure image port PR and cherry-picked on a new branch

**Comment by akshayben** [comment]
Haven't created 32 bit allocation. There is more work to do for this I believe (omrmem32helpers.c has the functions we have to mimic).

File: runtime/vm/segment.c [hunk 1]
```diff
@@ -130,10 +131,20 @@ void freeMemorySegment(J9JavaVM *javaVM, J9MemorySegment *segment, BOOLEAN freeD
 		} else if ((useAdvise) && (MEMORY_TYPE_JIT_SCRATCH_SPACE & segment->type)) {
 			j9mem_advise_and_free_memory(segment->baseAddress);
 		} else if (segment->type & (MEMORY_TYPE_RAM_CLASS | MEMORY_TYPE_UNDEAD_CLASS)) {
-			if (J9JAVAVM_COMPRESS_OBJECT_REFERENCES(javaVM)) {
-				j9mem_free_memory32(segment->baseAddress);
+			if (IS_COLD_RUN(javaVM)) {
+				if (J9JAVAVM_COMPRESS_OBJECT_REFERENCES(javaVM)) {
+					j9mem_free_memory32(segment->baseAddress); /* TODO: Create image memory free 32 bits */
```
Comment (tajila) [inline runtime/vm/segment.c]: The initial prototype is only intended to work on 64bit mode (no-compreseedrefs) so support for `j9mem_free_memory32` is not required. 

In fact I would add an assertion here (`Assert_VM_unreachable()`) and other places where `J9JAVAVM_COMPRESS_OBJECT_REFERENCES(javaVM)` is true on a cold run to make debugging easier.

Comment (tajila) [reply runtime/vm/segment.c]: actually, something like `assert(!"There is no support for compressedRefs");` is probably better

**Pull request merged**